lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -390,10 +391,13 @@ def datasets(self, request: Request, json: JSONOBJECT, query: Query) -> JSONOBJE
             try:
                 d["metadata"] = self._get_dataset_metadata(dataset, keys)
             except MetadataError as e:
+                errors.append(str(e))
```

Comment:
Are you sure that you want to compose a single list?  (Mightn't it grow uncomfortably large??)

Shouldn't we use the trick of building a counting `dict` with the error message as the key.